### DISCUSSION TITLE
Possible to provide a time estimate at which each layer occurs at?

### DISCUSSION THREAD

**Discussion by bryanseger** [sn4k3/UVtools#698] [Ideas]
I'm working on a project where I switch out the resin to a different material at a certain layer height. I am using UVtools to 'pause' the printer at this layer (by making the lift height incredibly high and by having a very long 'wait after retract' at this specific layer height).

It would be nice to know about how far along into the print this occurs, which spawned the idea of--can a time estimate be provided for each layer height?

**Comment by sn4k3** [comment]
It's possible to calculate that, however like total print time it's just an estimate which error depends on accelerations and if you depend on that you will miss the real layer, unless you have a huge wait time for that layer

**Reply by bryanseger** [reply]
The printers I'm working on, I have them dialed in to where the print estimate time that UVtools provides is usually within ~10 minutes of the actual print time on multi-hour prints. I know it's not going to be right on the exact second, but I feel that I have them close enough to where it'll be a useful tool for me to use.

I do have a 5 minute wait period at this particular layer. So if UVtools could give me an estimate print time occurrence for every layer, then that would give me a ballpark for setting a timer so I can prepare to change the vats over.

**Reply by sn4k3** [reply]
I can add more time properties in layers in order to obtain that.
For now you can divide total time per total layer and multiply by your layer number. It will give you an estimate.
Like, open UVtools terminal on file and send the command: `TimeSpan.FromSeconds(SlicerFile.PrintTime / SlicerFile.LayerCount * 300).ToString(@"hh\h\:mm\m")` where 300 is your desired layer number.

![image](https://user-images.githubusercontent.com/113281/236270349-0f28d062-e852-4824-a997-a9f3e0efa411.png)

**Reply by sn4k3** [reply]
With latest release you can do: `SlicerFile[layerno].StartTimeStr`

**Reply by bryanseger** [reply]
Thank you!
